spyder/plugins/pythonpath/container.py

```diff
@@ -74,195 +77,190 @@ def setup(self):
-        for path in self.project_path:
-            path_dict[path] = True
-
-        for path in self.path:
-            path_dict[path] = path not in self.not_active_path
-
-        return path_dict
-
-    def _update_python_path(self, new_path_dict=None):
         """
-        Update Python path on language server and kernels.
+        assert isinstance(user_paths, (type(None), OrderedDict))
+        assert isinstance(system_paths, (type(None), OrderedDict))
+        assert isinstance(prioritize, (type(None), bool))
 
-        The new_path_dict should not include the project path.
-        """
-        # Load existing path plus project path
-        old_path_dict_p = self._get_spyder_pythonpath_dict()
-
-        # Save new path
-        if new_path_dict is not None:
-            self._save_paths(new_path_dict)
-
-        # Load new path plus project path
-        new_path_dict_p = self._get_spyder_pythonpath_dict()
+        emit = False
 
-        # Do not notify observers unless necessary
-        if new_path_dict_p != old_path_dict_p:
-            pypath = self.get_spyder_pythonpath()
-            logger.debug(f"Update Pythonpath to {pypath}")
-            self.set_conf('spyder_pythonpath', pypath)
-            self.sig_pythonpath_changed.emit(old_path_dict_p, new_path_dict_p)
+        # Don't set options unless necessary
+        if user_paths is not None and user_paths != self._user_paths:
+            logger.debug(f"Saving user paths: {user_paths}")
+            self.set_conf('user_paths', dict(user_paths))
+            self._user_paths = user_paths
+
+        if system_paths is not None and system_paths != self._system_paths:
+            logger.debug(f"Saving system paths: {system_paths}")
+            self.set_conf('system_paths', dict(system_paths))
+            self._system_paths = system_paths
+
+        if prioritize is not None and prioritize != self._prioritize:
+            logger.debug(f"Saving prioritize: {prioritize}")
+            self.set_conf('prioritize', prioritize)
+            self._prioritize = prioritize
+            emit = True
+
+        spyder_pythonpath = self.get_spyder_pythonpath()
+        if spyder_pythonpath != self._spyder_pythonpath:
+            logger.debug(f"Saving Spyder pythonpath: {spyder_pythonpath}")
+            self.set_conf('spyder_pythonpath', spyder_pythonpath)
+            self._spyder_pythonpath = spyder_pythonpath
+            emit = True
+
+        # Only emit signal if spyder_pythonpath or prioritize changed
+        if emit:
+            self.sig_pythonpath_changed.emit(
+                self._spyder_pythonpath, self._prioritize
+            )
 
     def _migrate_to_config_options(self):
         """
         Migrate paths saved in the `path` and `not_active_path` files located
         in our config directory to our config system.
 
-        This was the way we save those paths in Spyder 5 and before.
+        ??? When should we remove this?
```

Comment:
We could remove it in Spyder 7, but people moving from Spyder 5 or 6.0.x to 7 would loose their saved paths. However, I don't think that's too bad because those versions would probably be at least two years old when 7 is released.

Reply:
That makes sense to me. What is the best way to go about that? A `TODO` comment? Or a PR with milestone set to 7?

Reply:
I think replacing this comment by a TODO one is a good solution.

Reply:
```suggestion
        # TODO: Remove for Spyder 7
```